### PR TITLE
Remove DOM component

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
This component was initially added for some early Notebook implementation but is no longer needed - and allowing extensions to insert arbitrary HTML directly into the DOM is a security risk.

This is a breaking change but given the severity of the issue and unlikeliness of someone else using this feature I see it as reasonable.

### CODE REVIEW

**Comment by chlafreniere** [comment]
Seems like an opportunity to add telemetry around component usage 😄

**Review by kburtram — APPROVED** [review]

File: src/sql/workbench/contrib/modelView/browser/components.contribution.ts [hunk 1]
```diff
@@ -107,9 +106,6 @@ registerComponentType(EDITOR_COMPONENT, ModelComponentTypes.Editor, EditorCompon
 export const DIFF_EDITOR_COMPONENT = 'diff-editor-component';
 registerComponentType(DIFF_EDITOR_COMPONENT, ModelComponentTypes.DiffEditor, DiffEditorComponent);
 
-export const DOM_COMPONENT = 'dom-component';
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/modelView/browser/components.contribution.ts:110]: we should also remove it from src\sql\platform\dashboard\browser\interfaces.ts

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/modelView/browser/components.contribution.ts:110]: Dang, good catch!

Comment (alanrenmsft) [reply src/sql/workbench/contrib/modelView/browser/components.contribution.ts:110]: and also from here :) https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/api/common/sqlExtHostTypes.ts#L173

**Review by alanrenmsft — APPROVED** [review]

**Review by chlafreniere — APPROVED** [review]

**Review by kisantia — APPROVED** [review]

**Pull request merged**